### PR TITLE
Add separators and direction to StackLayout

### PR DESCRIPTION
**Pull request by Fercas123**
This PR combines 2 old ones: 
https://github.com/jpmorganchase/salt-ds/pull/1040
https://github.com/jpmorganchase/salt-ds/pull/1037

to add horizontal direction to stack, and move separators from flex to stack.

 - Removed separators `div` wrapper with `display:content` from `FlexLayout`. 
   This change allows styles like `.classname > div` to be passed.
 - Add separators to `StackLayout`.
 - Add direction to `StackLayout` to allow horizontal stacks.

### CODE REVIEW

File: packages/core/src/stack-layout/StackLayout.tsx [hunk 1]
```diff
@@ -27,11 +46,34 @@ type StackLayoutComponent = <T extends ElementType = "div">(
...
+      children,
+      className,
+      direction = "column",
+      separators,
+      ...rest
+    }: StackLayoutProps<T>,
     ref?: PolymorphicRef<T>
   ) => {
+    const separatorAlignment = separators === true ? "center" : separators;
+    const flexDirection = useResponsiveProp(direction, "row");
```
Comment (el-dav) [inline packages/core/src/stack-layout/StackLayout.tsx:59]: Should the default be column?

File: packages/core/src/stack-layout/StackLayout.tsx [hunk 2]
```diff
@@ -10,14 +25,18 @@ export type StackLayoutProps<T extends ElementType> =
...
+      direction?: ResponsiveProp<LayoutDirection>;
       /**
        * Controls the space between items, default is 3.
        */
       gap?: FlexLayoutProps<ElementType>["gap"];
       /**
        * Adds a separator between elements, default is false.
        */
-      separators?: FlexLayoutProps<ElementType>["separators"];
+      separators?: LayoutSeparator | true;
```
Comment (el-dav) [inline packages/core/src/stack-layout/StackLayout.tsx:39]: Do you think seperators should also allow false? E.g. `<StackLayout seperators={false} />` seems valid e.g. if you want to conditionally toggle them on or off

File: packages/core/stories/stack-layout.doc.stories.mdx [hunk 3]
```diff
@@ -12,22 +12,36 @@ import HelpAndSupport from "docs/blocks/help-and-support.doc.mdx";
...
 
 <Canvas>
   <Story id="core-layout-stacklayout--default-stack-layout" />
 </Canvas>
 
 ## Use Stack layout when...
 
-You need a layout that is responsive by default and will break into two columns as soon as there is not enough space.
+- You need a layout that is consistently horizontal or vertical.
+- You need your items to be either scrollable or you want to use breakpoints for responsiveness.
```
Comment (el-dav) [inline packages/core/stories/stack-layout.doc.stories.mdx:28]: Maybe worth re-wording slightly to mention toggling between row/column via the responsive direction prop?

File: packages/core/stories/stack-layout.doc.stories.mdx [hunk 4]
```diff
@@ -12,22 +12,36 @@ import HelpAndSupport from "docs/blocks/help-and-support.doc.mdx";
...
 
 ### Don't use Stack layout when...
 
-You want the direction to have more complex customization. In this case we recommend you having a look at [Flex layout](?path=/docs/documentation-core-layout-flexlayout--page).
+- You want the direction to have more complex customization. In this case we recommend you having a look at [Flex layout](?path=/docs/documentation-core-layout-flexlayout--page).
```
Comment (el-dav) [inline packages/core/stories/stack-layout.doc.stories.mdx:32]: Maybe worth mentioning not to use when you want children to wrap as I think this is the key difference

File: packages/core/src/flex-layout/FlexLayout.css [hunk 5]
```diff
@@ -45,17 +49,17 @@
   top: 0;
 }
 .saltFlexLayout-separator-row-start > *:not(:last-child)::after {
-  right: calc(var(--flexLayout-gap) * 1);
+  right: 0;
 }
 .saltFlexLayout-separator-row-center > *:not(:last-child)::after {
-  right: 0;
+  right: calc(var(--flexLayout-gap) * -1);
```
Comment (el-dav) [inline packages/core/src/flex-layout/FlexLayout.css:55]: The seperator positioning doesn't quite seem to be working as expected in the stroybook. It seems to change it's position based on the combination of `gap` and `seperators="start"/"center"/"end"`

File: packages/core/src/stack-layout/StackLayout.tsx [hunk 6]
```diff
@@ -27,11 +46,42 @@ type StackLayoutComponent = <T extends ElementType = "div">(
 
 export const StackLayout: StackLayoutComponent = forwardRef(
   <T extends ElementType = "div">(
-    { children, ...rest }: StackLayoutProps<T>,
+    {
+      children,
+      className,
+      direction = "column",
+      gap,
```
Comment (el-dav) [inline packages/core/src/stack-layout/StackLayout.tsx:53]: changing 'gap' doesn't seem to be affecting the gap between items in the Stack. Does gap need to be passed to FlexLayout?

**Review by el-dav — APPROVED** [review]

File: packages/core/src/stack-layout/StackLayout.css [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+/* When alignment is 'row' and we have seperators (last child doesn't need a seperator) */
+.saltStackLayout-row.saltStackLayout-separator > *:not(:last-child)::after {
+  height: 100%;
+  width: var(--stackLayout-separator-weight);
+  right: var(--stackLayout-seperator-offset);
+  top: 0;
+}
+
+/* When alignment is 'row' and we have seperators (last child doesn't need a seperator) */
```
Comment (feliciackh) [inline packages/core/src/stack-layout/StackLayout.css:40]:
```suggestion
/* When alignment is 'column' and we have seperators (last child doesn't need a seperator) */
```

**Review by libertymayc — APPROVED** [review]

**Review by el-dav — APPROVED** [review]

**Pull request merged**